R/derive_params_growth_age.R

```diff
@@ -236,7 +238,10 @@ derive_params_growth_age <- function(dataset,
       by = c("sex_join", "ageu_join"),
       relationship = "many-to-many"
     ) %>%
-    filter(prev_age <= {{ age }} & {{ age }} < next_age)
+    mutate(age_diff := abs(metadata_age - {{ age }})) %>%
```

Comment:
@rossfarrugia I think this should do the trick, logic is as follows: 
1. calculate difference between current age provided, and the age provided in metadata (absolute value) 
2. calculate which record contains the smallest/minimum difference so in this case 25.49 should map to 25.5 not 24.5
3. filter by that record only

Reply:
@zdz2101 this update ensures that the nearest reference data is chosen, so it is an improvement but it doesn't solve completely what @CDC-DNPAO is asking us to do here which is to ensure that for every exact day of age we have a corresponding interpolated L/M/S to use instead of just matching to the nearest month when using CDC chart. So a patient who is say 748 days old doesn't use the 24.5 months L/M/S from CDC chart, they instead get an interpolated value which is more reflective of the fact they are actually in between 24.5 and 25.5 months.

Reflecting on this I think the place we need to do the interpolation is actually in our [ADVS template](https://github.com/pharmaverse/admiralpeds/blob/main/inst/templates/ad_advs.R) where we massage the reference files metadata into the form we need. We already convert months to days there for the CDC charts, but if we also then add the extra interpolated rows so that we have a record for every single day of age, then your function should work fine. FYI @Fanny-Gautier @Lina2689 @Minlei0201 

Let me pull some of us into a short meeting on this tomorrow so that we all agree the best plan forwards.

Reply:
Also @zdz2101 shouldn't the first step be to convert whatever age unit the user provides us with into days? e.g. if their age is calculated in months or weeks we'd still need to first convert to days as our reference metadata we are merging with is by days.

Reply:
@rossfarrugia The function was written to be unit-agnostic, as long as the input dataset's age units and the metadata's age units are **matching** , e.g. if you calculate current age is in months and you only have >= 2 yr olds in the study you don't need to modify the CDC data because it is already in months, if the current age calculation is in days and you are only working with the WHO criteria, or if you had some sort of hybridized columns of numeric of days and months as long as you have the appropriate metadata data with the correct range of days and months it should join fine